osgar/drivers/oak_camera.py

```diff
@@ -0,0 +1,158 @@
+
+            color_encoder.setDefaultProfilePreset(self.fps, dai.VideoEncoderProperties.Profile.MJPEG)
+
+            color.video.link(color_encoder.input)
+            color_encoder.bitstream.link(color_out.input)
+
+        if self.is_depth:
+            left = pipeline.create(dai.node.MonoCamera)
+            right = pipeline.create(dai.node.MonoCamera)
+            stereo = pipeline.create(dai.node.StereoDepth)
+            depth_out = pipeline.create(dai.node.XLinkOut)
+
+            queue_names.append("depth")
+            depth_out.setStreamName("depth")
+
+            left.setResolution(self.mono_resolution)
+            left.setBoardSocket(dai.CameraBoardSocket.LEFT)
+            left.setFps(self.fps)
+            right.setResolution(self.mono_resolution)
+            right.setBoardSocket(dai.CameraBoardSocket.RIGHT)
+            right.setFps(self.fps)
+
+            stereo.setDefaultProfilePreset(self.stereo_mode)
+            # https://docs.luxonis.com/projects/api/en/latest/components/nodes/stereo_depth/#currently-configurable-blocks
+            stereo.initialConfig.setMedianFilter(self.median_filter)
+            stereo.setExtendedDisparity(self.is_extended_disparity)
+            stereo.setSubpixel(self.is_subpixel)
+            stereo.setLeftRightCheck(self.is_left_right_check)  # https://docs.luxonis.com/en/latest/pages/faq/#left-right-check-depth-mode
+
+            left.out.link(stereo.left)
+            right.out.link(stereo.right)
+            stereo.depth.link(depth_out.input)
+
+        if not queue_names:
+            g_logger.error("No stream enabled!")
+            return
+
+        # TODO imu
+
+        if self.cam_ip and cam_is_available(self.cam_ip):  # USB cameras?
+            device_info = dai.DeviceInfo()
+            device_info.state = dai.XLinkDeviceState.X_LINK_BOOTLOADER
+            device_info.desc.protocol = dai.XLinkProtocol.X_LINK_TCP_IP
+            device_info.desc.name = self.cam_ip
+        else:
+            device_info = None
+            g_logger.info("Used the first available device.")
+
+        # Connect to device and start pipeline
+        with dai.Device(pipeline, device_info) as device:
+            while self.bus.is_alive():
+                queue_events = device.getQueueEvents(queue_names)
+
+                for queue_name in queue_events:
+                    if queue_name == "depth":
+                        packets = device.getOutputQueue(queue_name).tryGetAll()
+                        if len(packets) > 0:
+                            depth_frame = packets[-1].getFrame()  # use latest packet
+                            self.bus.publish("depth", depth_frame)
+
+                    if queue_name == "color":
+                        packets = device.getOutputQueue(queue_name).tryGetAll()
+                        if len(packets) > 0:
+                            color_frame = packets[-1].getData().tobytes()  # use latest packet
```

Comment:
what is the difference between `getData()` and `getFrame()` ... i.e. could it be "merged together"?

Reply:
JL - camera provides already JPEG images, i.e. not frame with width and height ... OK